app/Http/Controllers/Three/ReviewsController.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Rogue\Http\Controllers\Three;
+
+use Rogue\Models\Post;
+use Illuminate\Http\Request;
+use Rogue\Repositories\Three\PostRepository;
+use Rogue\Http\Controllers\Api\ApiController;
+use Illuminate\Auth\Access\AuthorizationException;
+use Rogue\Http\Transformers\Three\PostTransformer;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class ReviewsController extends ApiController
+{
+    /**
+     * The post service instance.
+     *
+     * @var Rogue\Repositories\ThreePostRepository
+     */
+    protected $post;
+
+    /**
+     * @var \Rogue\Http\Transformers\Three\PostTransformer
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @param  PostContract $posts
+     * @return void
+     */
+    public function __construct(PostRepository $post)
+    {
+        $this->post = $post;
+        $this->transformer = new PostTransformer;
+
+        $this->middleware('auth:api');
+    }
+
+    /**
+     * Update a post(s)'s status when reviewed.
+     *
+     * @param Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function reviews(Request $request)
+    {
+        $request->validate([
+            'post_id' => 'required',
+            'status' => 'in:pending,accepted,rejected',
+        ]);
+
+        // Only allow an admin to review the post.
+        if (token()->role() === 'admin') {
+            $review = $request->all();
+            $post = Post::where('id', $review['post_id'])->first();
+            $review['signup_id'] = $post->signup_id;
+            $review['northstar_id'] = $post->northstar_id;
+            $review['old_status'] = $post->status;
+
+            // Append admin's ID to the request for the "reviews" service.
+            $review['admin_northstar_id'] = auth()->id();
+            $reviewedPost = $this->post->reviews($review);
```

Comment:
I think we should consider moving away from passing arrays like this in our `Three` repositories/services. 

Instead, we can send named parameters for the things we care to know:

```php
$post = Post::find($request['post_id']);
$reviewedPost = $this->post->reviews($post, $request['status'], $request['comment']);
```


```php
/**
 * Updates a post's status after being reviewed.
 *
 * @param Post $post
 * @param string $status
 *
 * @return Post
 */
public function reviews(Post $post, $status, $comment)
{
    // Create the Review.
    $review = Review::create([
        'signup_id' => $post->signup_id,
        'northstar_id' => $post->northstar_id,
        'admin_northstar_id' => auth()->id(),
        'status' => $status,
        'old_status' => $post->status,
        'comment' => $comment,
        'post_id' => $post->id,
    ]);

    // Update the status on the Post.
    $post->status = $data['status'];
    $post->save();

    return $post;
}
```

This makes it clearer what the actual dependencies and inputs are at each step of the process.